debian/rules.in

```diff
@@ -55,18 +57,25 @@ override_dh_auto_build-arch:
 	dh_auto_build -- build-software
 
 override_dh_auto_build-indep:
+ifeq (,$(filter nodocs,$(DEB_BUILD_OPTIONS)))
 ifneq "$(enable_build_documentation)" ""
 	dh_auto_build -- manpages
 	dh_auto_build -- translateddocs
 	dh_auto_build -- docs
 endif
+endif
```

Comment:
Is it really a good idea to have two ways to set the same flag?  I suspect this will break the principle of least surprise.  What should happen if someone enable documentation building and use the nodocs option?  Which one should take effect?  I suspect it is better to only have one flag, and allow both the option and the enable setting to set or unset this flag.  At the moment this patch allow nodocs to override enable_documentation.  Is that least surprising?

Reply:
I suggest to ensure enable_build_documentation is "" if nodocs is set, and use only this test in the code, to ensure not two partly overlapping settings are used in the code.

Reply:
I just had a look at it again. The "enable_build_documentation" is set by debian/configure.in - and I agree with you, this should just always be enabled and the decision to build or not be left to Debian's regular semantics, i.e. the DEB_BUILD_OPTIONS parameters. The point of this patch is to help transitioning towards that change of control. So, now you can decide not to package the documentation (by adding the nodocs) even though the configuration (by debian/configure) was expecting you to build it.  The other way around does not work - you cannot enable the building of the documentation by omitting the nodocs if you have previously disabled it in debian/configure. I tend to think this is all very much how it should be - given the constraint to impose minimal changes to the current code base.

On a sidenote, at the beginning of debian/rules.in, there is
```
ifeq (,$(filter nodocs,$(DEB_BUILD_OPTIONS)))
enable_build_documentation = @ENABLE_BUILD_DOCUMENTATION@
endif
```
so enable_build_documentation will only be available if nodocs is not an option. I tend to think I am happy with how it is. I am uncertain if we should already use the nodocs/nocheck flags in our CI.